src/cryptonotecore/DatabaseBlockchainCache.cpp

```diff
@@ -283,7 +283,11 @@ namespace CryptoNote
         size_t requestPaymentIdTransactionsCount(IDataBase &database, const Crypto::Hash &paymentId)
         {
             auto batch = BlockchainReadBatch().requestTransactionCountByPaymentId(paymentId);
+#if defined (USE_LEVELDB)
             auto error = database.read(batch);
+#else
+            auto error = database.readThreadSafe(batch);
```

Comment:
Not sure where this one came from. It *might* be needed to be thread safe, but afaik, nothing actually consumes this corresponding API call. Any reason why we changed to `readThreadSafe` here?

Reply:
I think this one comes from @wrkzdev LevelDB, will check.

Reply:
it was the “use thread safe DB access for checkIfSpent” commit (yours, from Feb8 in TRTL)

https://github.com/turtlecoin/turtlecoin/commit/c75d4b5a9eb291f075ca14a25681d6d05a2aed6a

Reply:
That adds it to the `checkIfSpent` function, not `requestPaymentIdTransactionsCount`

Reply:
got it now, disregard the previous. This looks like @CapEtn’s add for levelDB, assuming based on the added if/else condition here

https://github.com/derogold/derogold/blob/dev-with-bin-files/src/cryptonotecore/DatabaseBlockchainCache.cpp#L286

I think he mentioned somewhere in the discord why he added it, I’ll ask him, don’t remember from the top of my head.

Reply:
hmmm. nope, still nope lol. I should go to bed. non-thread safe for levelDB, thread safe for RocksDB ... need to have a closer look.

Reply:
let’s just go for database.read(batch) here, for both RocksDB and LevelDB, changing it now.